code/modules/clothing/suits/armor.dm

```diff
@@ -194,4 +195,103 @@
 	name = "thunderdome suit"
 	desc = "Pukish armor."	//classy.
 	icon_state = "tdgreen"
-	item_state = "tdgreen"
\ No newline at end of file
+	item_state = "tdgreen"
+
+//LightToggle
+
+/obj/item/clothing/suit/armor/update_icon()
+	if(F)
+		if(F.on)
+			overlays += "flight-[initial(icon_state)]-on"
+		else
+			overlays += "flight-[initial(icon_state)]"
+	return
+
+/obj/item/clothing/suit/armor/ui_action_click()
+	toggle_gunlight()
+
+/obj/item/clothing/suit/armor/attackby(var/obj/item/A as obj, mob/user as mob, params)
+	if(istype(A, /obj/item/device/flashlight/seclite))
+		var/obj/item/device/flashlight/seclite/S = A
+		if(can_flashlight)
+			if(!F)
+				if(user.l_hand != src && user.r_hand != src)

```

Comment:
Yeah, I'm going to refine the code. I simply wanted to get the PR up firstly once I got it to work.